ApollosProject/apollos-prototype
# Adds ActionList algorithm driven by a content channel

## DESCRIPTION

### What does this PR do, or why is it needed?

Expands on previous work to add an algorithm that generates a content list driven by a ContentChannel in Rock. This allows churches to setup a content channel, and pull items for the action list directly from the content channel. This takes us one step away from wrapping up #800 

<img width="1814" alt="Screen Shot 2019-06-27 at 9 50 49 AM" src="https://user-images.githubusercontent.com/1637694/60271928-33ff7300-98c1-11e9-88d8-dec97bf555ef.png">


### What design trade-offs/decisions were made?

Haven't done the front end work yet.

### How do I test this PR?

1. Login via Graphiql
2. Run the following query, you should see two features
- one of them is the action list persona feature
- one of them is the action list content channel feature, which pulls content from here: 
https://apollosrock.newspring.cc/page/341?ContentChannelId=13

```
{
  userFeedFeatures {
    ... on ActionListFeature {
      id
      title
      subtitle
      actions {
        id
        title
        subtitle
        image {
          sources {
            uri
          }
        }
        relatedNode {
          id
        }
      }
    }
  }
}
```

### What automated tests did you write?

1. Datasource tests over all new functionality. 

## TODO:

- [x] References #800

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
